providers/cloudns/cloudnsProvider.go

```diff
@@ -13,24 +13,24 @@ import (
 )
 
 /*
-
 CloDNS API DNS provider:
-
 Info required in `creds.json`:
    - auth-id
    - auth-password
-
 */
 
 // NewCloudns creates the provider.
 func NewCloudns(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
 	c := &cloudnsProvider{}
 
-	c.creds.id, c.creds.password = m["auth-id"], m["auth-password"]
-	if c.creds.id == "" || c.creds.password == "" {
+	c.creds.id, c.creds.password, c.creds.subid = m["auth-id"], m["auth-password"],m["sub-auth-id"]
+	
+	if ((c.creds.id == "" && c.creds.subid == "") || c.creds.password == "" ) {
 		return nil, fmt.Errorf("missing ClouDNS auth-id and auth-password")
```

Comment:
Does this error message need to be updated?

Reply:
I thin yes, because I add "**sub-auth-id**"  in If loop to verify if sub-auth-id exist in my **creds.json** file